src/core/dnd/types.ts

```diff
@@ -59,76 +58,94 @@ export const canDropIntoLayout = (
 ) => {
   // check scope changes
   const detailContainer = getDetailContainer(layout);
-  if (!canDropIntoScope(item, detailContainer)) {
-    return false;
-  }
-  // check whether
-  return true;
+  return canDropIntoScope(item, detailContainer);
 };
 
 /**
  * Check whether the element to drop fits into the given scope,
  * e.g. whether a nested array object is dropped into the correct array ui schema control.
  *
  * @param item the drag and drop item
- * @param scopeChangingElement the nearest scope changing element,
+ * @param scopeUISchemaElement the nearest scope changing element,
  * e.g. the nearest array control into which shall be dropped.
  * Use `undefined` when dropping outside of any scope changing element.
  */
 export const canDropIntoScope = (
   item: NewUISchemaElement,
   scopeUISchemaElement: EditorUISchemaElement | undefined
 ) => {
-  // it's a control when there is a schema
-  if (item.schema) {
-    // check wether the control fits to the scope
-    // -- TODO check other cases than array
-    const scopeSchemaElement = getArrayContainer(item.schema);
-    if (!!scopeSchemaElement !== !!scopeUISchemaElement) {
-      // scopes don't match when one exist and the other doesn't
-      return false;
-    }
-    if (
-      scopeSchemaElement &&
-      scopeUISchemaElement &&
-      scopeUISchemaElement.linkedSchemaElement !== scopeSchemaElement.uuid
-    ) {
-      // scopes didn't match
+  const controlObject = item.schema;
+  if (controlObject) {
+    const scopeSchemaElement = getScopeChangingContainer(controlObject);
+    if (!scopesMatch(scopeSchemaElement, scopeUISchemaElement)) {
```

Comment:
can't we return `scopesMatch(scopeSchemaElement, scopeUISchemaElement)` ?

Reply:
We could, but should we add additional non-control-specific checks we must be careful to restore the current lines again. I would prefer the current "only return the cases we 100% know about approach", however I would also be fine with changing it.

Reply:
ok, fine by me